packages/vite-plugin-cloudflare/src/index.ts

```diff
@@ -905,4 +1015,33 @@ export function cloudflare(pluginConfig: PluginConfig = {}): vite.Plugin[] {
 			? resolvedPluginConfig.workers[environmentName]
 			: undefined;
 	}
+
+	/**
+	 * @returns Container options suitable for building or pulling images,
+	 * with image tag set to well-known dev format, or undefined if
+	 * containers are not enabled or not configured.
+	 */
+	async function getContainerOptions(
+		config: WorkerConfig,
+		containerBuildId: string
+	) {
+		if (!config.containers?.length || config.dev.enable_containers === false) {
+			return undefined;
+		}
+
+		const containers: ContainerDevOptions[] = [];
+		for (const container of config.containers) {
+			containers.push({
+				image: container.image ?? container.configuration?.image,
+				imageTag: getDevContainerImageName(
+					container.class_name,
+					containerBuildId
+				),
+				args: container.image_vars,
+				imageBuildContext: container.image_build_context,
+				class_name: container.class_name,
+			});
+		}
+		return containers;
```

Comment:
`map` seems more natural here:

```suggestion
		return config.containers.map((container) => {
			return {
				image: container.image ?? container.configuration?.image,
				imageTag: getDevContainerImageName(
					container.class_name,
					containerBuildId
				),
				args: container.image_vars,
				imageBuildContext: container.image_build_context,
				class_name: container.class_name,
			};
		});
```